pkg/operator/ceph/cluster/osd/key_rotation.go

```diff
@@ -0,0 +1,279 @@
+				},
+				TopologyKey: "kubernetes.io/hostname",
+			},
+		},
+	}
+}
+
+// getKeyRotationPodTemplateSpec returns the pod template spec for the key rotation job.
+func (c *Cluster) getKeyRotationPodTemplateSpec(osdProps osdProperties, osd OSDInfo, restart v1.RestartPolicy) (*v1.PodTemplateSpec, error) {
+	// create a volume on /dev so the pod can access devices on the host
+	devVolume := v1.Volume{Name: "devices", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/dev"}}}
+	udevVolume := v1.Volume{Name: "udev", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/run/udev"}}}
+	hostPathType := v1.HostPathDirectory
+	hostPath := filepath.Join(c.spec.DataDirHostPath, c.clusterInfo.Namespace, osdProps.pvc.ClaimName, fmt.Sprintf("ceph-%d", osd.ID))
+	hostPathVolume := v1.Volume{
+		Name: "bridge",
+		VolumeSource: v1.VolumeSource{
+			HostPath: &v1.HostPathVolumeSource{
+				Path: hostPath,
+				Type: &hostPathType,
+			},
+		},
+	}
+	devicesBasePath := "/var/lib/ceph/osd/"
+	volumes := []v1.Volume{
+		udevVolume,
+		devVolume,
+		hostPathVolume,
+	}
+	volumeMounts := []v1.VolumeMount{
+		{Name: "devices", MountPath: "/dev"},
+		{Name: "udev", MountPath: "/run/udev"},
+		{Name: "bridge", MountPath: devicesBasePath},
+	}
+
+	devices := []string{encryptionBlockDestinationCopy(devicesBasePath, bluestoreBlockName)}
+	if osdProps.metadataPVC.ClaimName != "" {
+		devices = append(devices, encryptionBlockDestinationCopy(devicesBasePath, bluestoreMetadataName))
+	}
+	if osdProps.walPVC.ClaimName != "" {
+		devices = append(devices, encryptionBlockDestinationCopy(devicesBasePath, bluestoreWalName))
+	}
+
+	keyRotationContainer, err := c.getKeyRotationContainer(osdProps, volumeMounts, devices)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to generate key rotation container")
+	}
+
+	podTemplateSpec := v1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: keyRotationCronJobAppName,
+			Labels: map[string]string{
+				k8sutil.AppAttr:     keyRotationCronJobAppName,
+				k8sutil.ClusterAttr: c.clusterInfo.Namespace,
+			},
+			Annotations: map[string]string{},
+		},
+		Spec: v1.PodSpec{
+			ServiceAccountName: serviceAccountName,
+			Containers: []v1.Container{
+				keyRotationContainer,
+			},
+			RestartPolicy:     restart,
+			Volumes:           volumes,
+			HostNetwork:       c.spec.Network.IsHost(),
+			PriorityClassName: cephv1.GetOSDPriorityClassName(c.spec.PriorityClassNames),
+			SchedulerName:     osdProps.schedulerName,
+		},
+	}
+	if c.spec.Network.IsHost() {
+		podTemplateSpec.Spec.DNSPolicy = v1.DNSClusterFirstWithHostNet
+	} else if c.spec.Network.IsMultus() {
+		if err := k8sutil.ApplyMultus(c.spec.Network, &podTemplateSpec.ObjectMeta); err != nil {
+			return nil, err
+		}
+	}
+
+	// Replace default unreachable node toleration if the osd pod is portable and based in PVC
+	if osdProps.portable {
+		k8sutil.AddUnreachableNodeToleration(&podTemplateSpec.Spec)
```

Comment:
> We shouldn't need this toleration for the cron job, just for daemon pods

I think it should apply to key rotation pod since it applies to OSD as well.
we want this pod to be eviction duration to be similar to that of OSD pod's.

Reply:
removed it :+1: